appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -163,25 +155,27 @@ internal class ExperienceRenderer(
         }
     }
 
-    suspend fun start(owner: StateMachineOwning, context: RenderContext) {
+    suspend fun start(frame: AppcuesFrameView, context: RenderContext) {
         // If there's already a frame for the context, reset it back to its unregistered state.
         stateMachines.getOwner(context)?.reset()
 
-        // If the machine being started is already registered for a different context,
-        // reset it back to its unregistered state before potentially showing new content.
-        owner.reset()
```

Comment:
this was never doing anything, since we were passing a new `owner` in each time, and it never had a state machine.

the new and improved check will actually look for the case of the same `frame` already being registered in a different context, and reset that old context if so - since the new registration should be unique for this frame.